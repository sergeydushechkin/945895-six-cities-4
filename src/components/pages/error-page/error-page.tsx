import * as React from "react";
import {connect} from "react-redux";

import {getErrorText} from "../../../reducer/data/selectors";

import Header from "../../header/header";

interface Props {
  errorText: string;
}

const ErrorPage: React.FunctionComponent<Props> = (props: Props) => {
  const {errorText} = props;

  return (
    <div className="page page--gray page--main page__main--index-empty">
      <Header
        isLogoActive = {true}
      />
      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">
            <ul className="locations__list tabs__list">
            </ul>
          </section>
        </div>
        <div className="cities">
          <div className="cities__places-container cities__places-container--empty container">
            <section className="cities__no-places">
              <div className="cities__status-wrapper tabs__content">
                <b className="cities__status">{errorText}</b>
              </div>
            </section>
            <div className="cities__right-section"></div>
          </div>
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    errorText: getErrorText(state),
  };
};

export {ErrorPage};
export default connect(mapStateToProps, null)(ErrorPage);
