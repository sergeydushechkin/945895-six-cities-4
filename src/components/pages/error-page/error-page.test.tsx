import * as React from "react";
import * as renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";

import history from "../../../history";
import {ErrorPage} from "./error-page";

import testStore from "../../../mocks/tests-mock-store";

const mockStore = configureStore([]);

it(`Render ErrorPage`, () => {
  const store = mockStore(testStore);

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <ErrorPage errorText={`Test error`}/>
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
