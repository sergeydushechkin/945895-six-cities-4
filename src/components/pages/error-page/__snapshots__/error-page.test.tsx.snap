// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ErrorPage 1`] = `
<div
  className="page page--gray page--main page__main--index-empty"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              Test error
            </b>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
